### PR TITLE
HARP-8498: Support picking with terrain.

### PR DESCRIPTION
**Pull request by atomicsulfate**
Signed-off-by: Andres Mandado <andres.mandado-almajano@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/lib/PickHandler.ts [hunk 1]
```diff
@@ -101,18 +102,22 @@ export interface PickResult {
...
  */
 export class PickHandler {
+    readonly rayCaster: PickingRaycaster;
     private readonly m_plane = new THREE.Plane(new THREE.Vector3(0, 0, 1));
     private readonly m_roadPicker?: RoadPicker;
 
     constructor(
         readonly mapView: MapView,
         readonly camera: THREE.Camera,
+        width: number,
```
Comment (ninok) [inline @here/harp-mapview/lib/PickHandler.ts]: Do we recreate the PickHandler when we resize?

Comment (atomicsulfate) [reply @here/harp-mapview/lib/PickHandler.ts]: No, we don't, I already saw we have a bug there

File: @here/harp-mapview/lib/PickHandler.ts [hunk 2]
```diff
@@ -232,6 +247,46 @@ export class PickHandler {
         return pickResults;
     }
 
+    private intersectMapObjects2D(x: number, y: number, intersects: THREE.Intersection[]) {
+        // Find 2D objects at the picking ray intersection on the ground.
+        const rayCaster = this.rayCaster;
+        rayCaster.setFromCamera(
+            this.mapView.getNormalizedScreenCoordinates(x, y),
+            this.mapView.camera
```
Comment (nzjony) [inline @here/harp-mapview/lib/PickHandler.ts]: this should be the rteCamera

File: @here/harp-mapview/lib/PickHandler.ts [hunk 3]
```diff
@@ -232,6 +247,46 @@ export class PickHandler {
...
+        rayCaster.set(
+            this.camera.position,
+            worldGroundIntersect.sub(this.mapView.camera.position).normalize()
+        );
+
+        // for (const tile of intersectTiles) {
+        //     rayCaster.intersectObjects(tile.objects, true, intersects);
+        // }
+
+        for (const object of this.mapView.worldRootObject.children) {
```
Comment (nzjony) [inline @here/harp-mapview/lib/PickHandler.ts]: do you need the loop here? If you pass recursive to intersectObject it should do it internally ?

**Pull request closed without merging**